Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting_test.py

```diff
@@ -0,0 +1,1010 @@
+])
+def test_pagination_wrong_input(page, page_size, expected_err_msg):
+    """
+    Tests the pagination function.
+
+        Given:
+            1+2 -  page and page size arguments with 0 value.
+            3+4 -  page and page size arguments with < 0 value.
+
+        When:
+            - Running the 'pagination function'.
+
+        Then:
+            - Checks that the expected err message is raised.
+    """
+    with pytest.raises(DemistoException) as e:
+        pagination(page, page_size)
+    assert e.value.args[0] == expected_err_msg
+
+
+@pytest.mark.parametrize('raw_response', [DESTINATION_LIST_RESPONSE])
+def test_test_module(requests_mock, raw_response):
+    """
+        Tests the test_module function.
+
+            Given:
+                - no argument required.
+
+            When:
+                - Running the 'test_module function'.
+
+            Then:
+                - Check weather the given credentials are correct or not.
+    """
+    from CiscoUmbrellaReporting import test_module
+    post_req_url = client.token_url
+    requests_mock.post(post_req_url, json={'access_token': '12345'})
+    access_token = client.get_access_token()
+    get_req_url = f'{client._base_url}/v2/organizations' \
+                  f'/{client.organisation_id}/activity'
+    headers = {'Authorization': f'Bearer {access_token}'}
+    requests_mock.get(get_req_url, headers=headers, json=raw_response)
+    output = test_module(client)
+    assert output == 'ok'
+    with pytest.raises(DemistoException):
+        error_output = {
+            "meta": {},
+            "data": {
+                "error": "unauthorized"
+            }
+        }
+        requests_mock.get(get_req_url, headers=headers, status_code=401,
+                          json=error_output)
+        test_module(client)
+
+
+def test_access_token(requests_mock):
+    """
+        Tests the access_token function.
+
+            Given:
+                - requests_mock object.
+
+            When:
+                - Running the 'get_activity_list_command function'.
+
+            Then:
+                -  Checks the output of the command function with the expected output.
+    """
+    req_url = client.token_url
+
+    requests_mock.post(req_url, json={'access_token': '12345'})
+
+    response = client.get_access_token()
+    assert response == "12345"
+
+    with pytest.raises(DemistoException) as e:
+        requests_mock.post(req_url, status_code=401)
+        client.get_access_token()
+    assert e.value.args[0] == ('Authorization Error: The provided credentials for Cisco Umbrella Reporting are invalid.'
+                               ' Please provide a valid Client ID and Client Secret.')
+
+    with pytest.raises(DemistoException) as e:
+        requests_mock.post(req_url, status_code=400)
+        client.get_access_token()
+    assert e.value.args[0] == 'Error: something went wrong, please try again.'
+
+
+def test_check_valid_indicator_value():
```

Comment:
Please split it into 2 separate tests: 
1. Tests the valid cases - valid indicators. Please test all the possible cases including list of indicators. 

2. Tests the invalid cases - invalid indicators. Please use parameterized to test all possible cases: invalid domain, invalid IP .... and the expected error messages. 

You can use the new pagination tests I added as a reference.

Reply:
@ShacharKidor Done